server/routes/analyticsRecommendation.js

```diff
@@ -1,28 +1,94 @@
-// Delivers one recommendation banner per request, based on
-// journal + mood streak analysis in utils/recommendation.js
-
 const express = require('express');
 const { STATUS } = require('../constants');
 const checkAuth = require('../middleware/checkAuth');
-const getRecommendation = require('../utils/recommendation');
+const { PrismaClient } = require('@prisma/client');
+const prisma = new PrismaClient();
+const { getRankedBanners } = require('../utils/recEngine');
+
 
 const router = express.Router();
 router.use(checkAuth);
 
 // GET /api/recommendation
-// Reads the current user's streak metrics and returns the best-fit banner
-router.get('/recommendation', async (req, res) => {
+router.get('/', async (req, res) => {
   try {
-    const banner = await getRecommendation(req.userId);
+    // Get all ranked banners for this user
+    const banners = await getRankedBanners(req.userId);
+
+    // Get the top banner (if any)
+    const topBanner = banners[0] || null;
+
+    // If we have a banner to show, record that it was seen
+    if (topBanner) {
+      await prisma.userBannerHistory.upsert({
+        where: {
+          userId_bannerTag: {
+            userId: req.userId,
+            bannerTag: topBanner.tag,
+          },
+        },
+        update: {
+          seenAt: new Date(),
+        },
+        create: {
+          userId: req.userId,
+          bannerTag: topBanner.tag,
+          seenAt: new Date(),
+        },
```

Comment:
As we discussed live, it is misleading to record seenAt when server sends a banner to show. It should be recorded from the client when it is actually shown